File: TMessagesProj/src/main/java/tw/nekomimi/nekogram/utils/AlertUtil.java
package tw.nekomimi.nekogram.utils;
import android.app.Activity;
import android.content.Context;
import org.telegram.ui.ActionBar.AlertDialog;
import org.telegram.ui.ActionBar.BaseFragment;
public class AlertUtil {
    public static void showToast(String msg) {}
    public static void showToast(Throwable t) {}
    public static void showToast(Exception e) {}
    public static void showToast(Object obj) {}
    public static void showSimpleAlert(BaseFragment f, String title, String msg) {}
    public static void showSimpleAlert(Object f, String msg) {}
    public static void copyAndAlert(String text) {}
    public static AlertDialog showProgress(Activity a) { return null; }
    public static AlertDialog showProgress(Context c) { return null; }
    public static void showTransFailedDialog(Activity a, boolean unsupported, String message, Runnable onRetry) {}
    public static void showTransFailedDialog(Context c, boolean unsupported, String message, Runnable onRetry) {}
    public static void showConfirm(Activity a, String title, int icon, String message, boolean red, Runnable onConfirm) {}
    public static void showCopyAlert(Activity a, String text) {}
}

File: TMessagesProj/src/main/java/tw/nekomimi/nekogram/utils/AlertUtil.kt
package tw.nekomimi.nekogram.utils

import android.content.Context
import org.telegram.tgnet.TLRPC
import org.telegram.ui.ActionBar.AlertDialog

object AlertUtil {

    @JvmStatic
    fun copyAndAlert(text: String) {}

    @JvmStatic
    fun copyLinkAndAlert(text: String) {}

    @JvmStatic
    fun call(number: String) {}

    @JvmStatic
    fun showToast(e: Throwable) {}

    @JvmStatic
    fun showToast(e: TLRPC.TL_error?) {}

    @JvmStatic
    fun showToast(text: String) {}

    @JvmStatic
    fun showSimpleAlert(ctx: Context?, error: Throwable) {}

    @JvmStatic
    @JvmOverloads
    fun showSimpleAlert(ctx: Context?, text: String, listener: ((AlertDialog.Builder) -> Unit)? = null) {}

    @JvmStatic
    @JvmOverloads
    fun showSimpleAlert(ctx: Context?, title: String?, text: String, listener: ((AlertDialog.Builder) -> Unit)? = null) {}

    @JvmStatic
    fun showCopyAlert(ctx: Context, text: String) {}

    @JvmOverloads
    @JvmStatic
    fun showProgress(ctx: Context, text: String = "Loading"): AlertDialog {
        return AlertDialog.Builder(ctx).create()
    }

    fun showInput(ctx: Context, title: String, hint: String, onInput: (AlertDialog.Builder, String) -> String) {}

    @JvmStatic
    @JvmOverloads
    fun showConfirm(ctx: Context, title: String, text: String? = null, icon: Int, button: String, red: Boolean, listener: Runnable) {}

    @JvmStatic
    @JvmOverloads
    fun showTransFailedDialog(ctx: Context, noRetry: Boolean, message: String, retryRunnable: Runnable) {}

    fun showTimePicker(ctx: Context, title: String, callback: (Long) -> Unit) {}
}
